### DISCUSSION TITLE
Add flexibility for Breadcrumbs to use history API navigation

### DISCUSSION THREAD

**Discussion by vankhoawin** [twilio-labs/paste#1588] [Ideas] [answered]
https://paste.twilio.design/components/breadcrumb/

Hi team, question about the Breadcrumb component. We are using it just fine in our application, but it would be even more consistent with our usual Anchors if we could also make Breadcrumb links use JavaScript-based navigation to avoid a full page render. Would it be possible to hook into the `BreadcrumbItem` component to give this functionality?

**Comment by TheSisb** [comment, marked as answer]
Hi @vnguyen94,

I shared this suggestion with my team and we all agree it is something we should support.  I've created a ticket in our backlog to move the prop spreading from the wrapping `<Box as="li">` to the child element (Anchor or Text).  
So moving [line 32](https://github.com/twilio-labs/paste/blob/main/packages/paste-core/components/breadcrumb/src/index.tsx#L32) to line 41 and 45.  What do you think about that approach?

**Reply by vankhoawin** [reply]
We're using a copy of the React Router pattern from the Anchor documentation: https://paste.twilio.design/components/anchor/#using-anchor-with-a-router

In this example, an entirely new component is used instead of a plain Anchor. I don't wanna put any solutions on the board myself, but only allowing props to be set on the Anchor might hinder other routing solutions from being able to hook into the Breadcrumbs component.

**Reply by TheSisb** [reply]
We'll keep that particular point in mind. Thanks for the feedback!

**Comment by jweintraub** [comment]
Please note that in the Console, individual product teams won't be implementing breadcrumbs - the breadcrumbs are being implemented now by the Console team.

**Reply by vankhoawin** [reply]
While that is true, we're also using Breadcrumbs (and Paste in general) for internal app use. The great thing about Paste is that any application (even non-Twilio) can use the design system, so it'd be cool to be able to use it in a variety of different environments.
